util/dvsim/SlurmLauncher.py

```diff
@@ -0,0 +1,146 @@
+# Copyright lowRISC contributors (OpenTitan project).
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+import os
+import shlex
+import subprocess
+
+from Launcher import ErrorMessage, Launcher, LauncherError
+
+
+SLURM_QUEUE = os.environ.get("SLURM_QUEUE", "hw-m")
+SLURM_MEM = os.environ.get("SLURM_MEM", "16G")
+SLURM_MINCPUS = os.environ.get("SLURM_MINCPUS", "8")
+SLURM_TIMEOUT = os.environ.get("SLURM_TIMEOUT", "240")
+SLURM_CPUS_PER_TASK = os.environ.get("SLURM_CPUS_PER_TASK", "8")
+SLURM_SETUP_CMD = os.environ.get("SLURM_SLURM_SETUP_CMD", "module purge; module load opentitan")
+
+
+class SlurmLauncher(Launcher):
+    # Misc common SlurmLauncher settings.
+    max_odirs = 5
+
+    def __init__(self, deploy):
+        '''Initialize common class members.'''
+
+        super().__init__(deploy)
+
+        # Popen object when launching the job.
+        self.process = None
+
+    def _do_launch(self):
+        global job_name
+        # Update the shell's env vars with self.exports. Values in exports must
+        # replace the values in the shell's env vars if the keys match.
+        exports = os.environ.copy()
+        if self.deploy.exports:
```

Comment:
Is the `if` needed here?
```
Python 3.12.7 (main, Oct  3 2024, 15:15:22) [GCC 14.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> a = {1: 1, 2: 2, 3: 3}
>>> a.update({})
>>> a
{1: 1, 2: 2, 3: 3}
```

(or is the point that it's `None` or something? In which case maybe `if self.deploy.exports is not None` ?)

Reply:
I was using the same code pattern as in other launchers. I traced it down, `deploy.exports` is coming from [here](https://github.com/lowRISC/opentitan/blob/master/util/dvsim/Deploy.py#L213) and always at least defines an empty dict.

What about a second commit, removing the `if` for all launchers?

Reply:
That's perfect: thanks!